primitives/state-machine/src/testing.rs

```diff
@@ -348,4 +393,32 @@ mod tests {
 		ext.commit_all().unwrap();
 		assert!(ext.backend.eq(&backend), "Both backend should be equal.");
 	}
+
+	#[test]
+	fn execute_and_generate_proof_works() {
```

Comment:
I tried to make this test such that I would make two `ext`s with the same initial data, read 1 key from one and 3 keys from the other one, and expect the proof of the latter to be bigger than the former, but this failed. Any idea why @cheme?

Reply:
If value are small, they can be encoded in a same node (inline encoding) and the proof will be the same length.
Using values of length 32 or more is better for testing this kind of things.

Reply:
Sounds like a bad test.

Convert the proof back to a db and check that it contains the expected values.

Reply:
the backing memory db contains the same data trie nodes of `proof`, and thei hash, not sure what is there to check. If you want me to improve the test, doing in on the `proof` itself is enough. 

I am looking into checking the proof itself, but it sounds a bit like an overkill. It is not the scope of this test to make sure the proof recorded is doing its job well, just that _something_ is recorded. 

Inserting large values (32+ bytes) does help indeed.

Reply:
looking a bit into example code, I presume making sure that the proof (as memory-db or as-is) contains the root is a reasonable check to do.